### PR TITLE
feat: [SSCA-671]: Regex support for PackageName in Deny List

### PR DESCRIPTION
**Pull request by deepharness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53679)

### CODE REVIEW

**Comment by deepharness** [comment]
trigger uts

File: ssca-manager/service/src/main/java/io/harness/ssca/enforcement/executors/mongo/MongoInterpreter.java [hunk 1]
```diff
@@ -18,10 +18,7 @@
 import io.harness.ssca.enforcement.executors.mongo.filter.allowlist.PurlQueryBuilder;
 import io.harness.ssca.enforcement.executors.mongo.filter.allowlist.SupplierQueryBuilder;
 import io.harness.ssca.enforcement.executors.mongo.filter.denylist.DenyListItemQueryBuilder;
-import io.harness.ssca.enforcement.executors.mongo.filter.denylist.fields.LicenseField;
-import io.harness.ssca.enforcement.executors.mongo.filter.denylist.fields.PurlField;
-import io.harness.ssca.enforcement.executors.mongo.filter.denylist.fields.SupplierField;
-import io.harness.ssca.enforcement.executors.mongo.filter.denylist.fields.VersionField;
+import io.harness.ssca.enforcement.executors.mongo.filter.denylist.fields.*;
```
Comment (inderschera) [inline ssca-manager/service/src/main/java/io/harness/ssca/enforcement/executors/mongo/MongoInterpreter.java:21]: lets fix * imports

**Review by deepharness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53679)* status: 2 file(s) to review. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`ssca-manager/service/src/main/java/io/harness/ssca/enforcement/executors/mongo/MongoInterpreter.java` line 21 at r1](https://reviewable.io/reviews/harness/harness-core/53679#gh-1345197457:-NftS_lf3NIChN6rJtlb:b-7rl0kd) ([raw file](https://github.com/harness/harness-core/blob/1e44946bb47964f26c6239afee728a9b783d5abb/ssca-manager/service/src/main/java/io/harness/ssca/enforcement/executors/mongo/MongoInterpreter.java#L21)):*
<details><summary><i>Previously, inderschera (Inderpreet Chera) wrote…</i></summary><blockquote>

lets fix * imports
</blockquote></details>

done

**Comment by deepharness** [comment]
trigger uts

**Comment by inderschera** [comment]
trigger uts

**Pull request merged**